keras/src/layers/preprocessing/image_preprocessing/random_apply.py

```diff
@@ -0,0 +1,279 @@
+    ```
+    """
+
+    def __init__(
+        self,
+        layer,
+        rate=0.5,
+        batchwise=False,
+        auto_vectorize=False,
+        seed=None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if not (0 <= rate <= 1.0):
+            raise ValueError(
+                f"rate must be in range [0, 1]. Received rate: {rate}"
+            )
+        self._layer = layer
+        self._rate = rate
+        self.auto_vectorize = auto_vectorize
+        self.batchwise = batchwise
+        self.seed = seed
+        self.built = True
+        if K.backend() == "jax":
+            self.seed_generator = random.SeedGenerator(seed)
+
+    def _get_should_augment(self, inputs, seed=None):
+        input_shape = ops.shape(inputs)
+
+        if self.batchwise:
+            return self._rate > random.uniform(shape=(), seed=seed)
+
+        batch_size = input_shape[0]
+        random_values = random.uniform(shape=(batch_size,), seed=seed)
+        should_augment = random_values < self._rate
+
+        ndims = len(inputs.shape)
+        ones = [1] * (ndims - 1)
+        broadcast_shape = tuple([batch_size] + ones)
+
+        return ops.reshape(should_augment, broadcast_shape)
+
+    def _augment_single(self, inputs, seed=None):
+        random_value = random.uniform(shape=(), seed=seed)
+        should_augment = random_value < self._rate
+
+        def apply_layer():
+            if hasattr(self._layer, "get_random_transformation"):
+                transformation = self._layer.get_random_transformation(
+                    inputs, training=True, seed=seed
+                )
+                return self._layer.transform_images(
+                    inputs, transformation, training=True
+                )
+            return self._layer(inputs)
+
+        def return_inputs():
+            return inputs
+
+        return ops.cond(should_augment, apply_layer, return_inputs)
+
+    def _augment_batch(self, inputs, seed=None):
+        should_augment = self._get_should_augment(inputs, seed=seed)
+
+        if hasattr(self._layer, "get_random_transformation"):
+            transformation = self._layer.get_random_transformation(
+                inputs, training=True, seed=seed
+            )
+            augmented = self._layer.transform_images(
+                inputs, transformation, training=True
+            )
+        else:
+            augmented = self._layer(inputs)
+
+        return ops.where(should_augment, augmented, inputs)
+
+    def call(self, inputs):
+        if isinstance(inputs, dict):
+            return {
+                key: self._call_single(input_tensor)
+                for key, input_tensor in inputs.items()
+            }
+        else:
+            return self._call_single(inputs)
+
+    def _call_single(self, inputs):
+        inputs_rank = len(inputs.shape)
+        is_single_sample = ops.equal(inputs_rank, 3)
+        is_batch = ops.equal(inputs_rank, 4)
+
+        if K.backend() == "jax":
```

Comment:
This is not the way to do it -- please check out other preprocessing layers for an example of how to handle the seed generator and seed argument